pkg/maps/metricsmap/metricsmap.go

```diff
@@ -183,23 +177,88 @@ func SyncMetricsMap() error {
 
 	var key, nextKey Key
 	for {
-		err := metricsmap.GetNextKey(&key, &nextKey)
+		err := bpf.GetNextKey(metricsmap.GetFd(), unsafe.Pointer(&key), unsafe.Pointer(&nextKey))
 		if err != nil {
 			break
 		}
-		entry, err := metricsmap.Lookup(&nextKey)
+		err = bpf.LookupElement(metricsmap.GetFd(), unsafe.Pointer(&nextKey), unsafe.Pointer(&entry))
 		if err != nil {
 			return fmt.Errorf("unable to lookup metrics map: %s", err)
 		}
-		value := entry.(*Value)
-		// Increment Prometheus metrics here.
-		updatePrometheusMetrics(&nextKey, value)
-		key = nextKey
+
+		// cannot use `range entry` since, if the first value for a particular
+		// CPU is zero, it never iterates over the next non-zero value.
+		for i := 0; i < possibleCpus; i++ {
+			// Increment Prometheus metrics here.
+			updatePrometheusMetrics(&nextKey, &entry[i])
```

Comment:
Is this guaranteed to be a local cache or could it reach out over the network? If the latter, it might be better to aggregate first and perform one update for all CPUs.

Reply:
This simply updates the local cache. The actual HTTP request to prometheus server simply uses this cached value and sends it over the network. So we should be ok.